.github/workflows/release-publish.yml

```diff
@@ -2,21 +2,67 @@ name: release-publish
 
 on:
   push:
-    tags:
-      - '*-v*'
+    branches:
+      - main
+
+permissions:
+  contents: write # Required for pushing the Helm charts to the gh-pages branch
+  packages: write # Required for GHCR access
+  id-token: write # Required for signing
 
 jobs:
   release:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
-          fetch-depth: 0
+          fetch-depth: 0 # important for fetching all history to run comparison against
+
+      - name: Fetch history
+        run: git fetch --prune
 
-      - name: Publish Helm charts
-        uses: stefanprodan/helm-gh-pages@0ad2bb377311d61ac04ad9eb6f252fb68e207260 # v1.7.0
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Set up Helm
+        uses: azure/setup-helm@29960d0f5f19214b88e1d9ba750a9914ab0f1a2f # v4.0.0
         with:
-          helm_version: 3.4.0
-          token: "${{ secrets.REPO_GHA_PAT }}"
+          version: v3.14.1
+
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@a917fd15b20e8b64b94d9158ad54cd6345335584 # v1.6.0
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          CR_GENERATE_RELEASE_NOTES: true
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: sigstore/cosign-installer@e1523de7571e31dbe865fd2e80c5c7c23ae71eb4 # v3.4.0
```

Comment:
This please for sure requires a `-name` and at least a description! it's invisible!

Reply:
Yhea. Thanks. I'll do that.